articles/0064/_posts/2024-06-11-0064-ChoosingRubyToTeachProgramming.md

```diff
@@ -0,0 +1,233 @@
+---
+layout: post
+title: 自分の経験からRubyでプログラミングを教えた
+short_title: My Experience on Teaching Programming with Ruby
+tags: 0064
+post_author: ホアンクアン
+created_on: 2024 年 6 月 11 日
+---
+
+{% include base.html %}
+
+# 日本語バージョン：
+
+## はじめに
+
+IT業界全般、特にプログラミングは、人材育成は大事なことです。ですが、新しいメンバーをトレーニングするとき、メンターとしてエンジニアは、様々なチャレンジに会います。この投稿は、私の経験や、なぜRubyを選んでプログラミングを教えているかの理由です。よろしくお願いします。
+
+## 自己紹介
+
+初めまして。
+
+ベトナムのハノイ出身のホアン・クアンです。４年間日本で働いています。今はリードエンジニアとして東京のITスタートアップで働いています。好きな技術スタックはAgile/Scrum、AWSとRubyです。
+
+最初の１年半、Rubyプログラマーでした。その後、システムエンジニアとして仕事しており、仕事の内容はあまりRubyを使っていませんが、自分でRubyを勉強し続けています。
+
+よろしくお願いします。
+
+## 状況
+
+会社設立当初、CEOが知り合いをインターンやバイトとして雇いたいということで、リードエンジニアである私は彼らの育成も任されました。CEOの知り合いは、6人の若いベトナム留学生でした。彼らは、卒業後、IT関連の仕事をやりたかったので、私はスキルと経験を伝えたかったです。しかし、...
+
+### チャレンジ#1
+
+私のケースは、これらの問題に合いました：
+1. 留学生たちは日本語学校やIT専門学校で勉強していましたが、その学校でITやプログラミングをあまり勉強していませんでした。
+1. IT関連の仕事をやりたいですが、彼らは具体的に何をするが分からなかったです。
+1. 最近、私のベトナムコミュニティーに、「プログラミングできず、IT仕事もできる」というコースもあります。というのは、ベトナムのアウトソース会社は、3ヶ月から6ヵ月ぐらい、ITを知らない人にBrSE（ブリッジSE）になるためことを教えることです。そのことは、だいたいビジネスアナリスト(BA)の知識ですから、プログラミングは全然ないです。すなわち、留学生たちとして、「簡単より道」です。
+1. 英語が分からなくて、日本語能力もまだ強くなかったです。
+1. 問題があったら、資料や本を読まずに、ネットで検索せずに、すぐに聞きます。
+1. 育成期間は3ヶ月だけでした。
+1. プログラミング言語が多いです。
+1. シラバスがまだなかったです。
+
+上記のことから、解決ために、適切なプログラミング言語を選択し、それを使用して週ごとにシラバスを作成し、他のソリューションと組み合わせる必要がありました。
+
+### チャレンジ#1の解決
+私の対策は:
+- 1、6、7の問題は、シンタックスが理解しやすく、早く勉強できるプログラミング言語を選びました。
+- 2の問題は、彼たちへオリエンテーションを企画しました。
+- 3の問題は、選んだプログラミング言語とオリエンテーションでした。プログラミング言語ソリューションに関しては、他の短期コースと競合するのに十分な速さで学習できる言語を選択する必要がありまsた。
+- 4、5の問題は、彼たちに[「Read - Search - Ask」](https://www.freecodecamp.org/news/read-search-dont-be-afraid-to-ask-743a23c411b4/)を教えるし、参考資料をちゃんと準備しました。
+- 8の問題は、週ごとにシラバスを作成しなければしかありませんでした。
+
+この投稿に限り、**「シンタックスが理解しやすく、早く勉強できるプログラミング言語を選ぶ」**方法をもっと続きたいです。Ruby、PythonやTypescriptは、学習が簡単ですぐに習得できるというこの要件を満たしています。
+
+このとき、Pythonを選びました。理由は：
+- Rubyのシンタックスを似ています。
+- AWSに働くとき、Lambda、Glue ETLのようなサービスに、Pythonを使うことがたくさんです。ですから、私も経験もあります。
```

Comment:
`When I work at AWS,` は、 AWS に就職をして仕事をしていたときというニュアンスに感じますが、あっていますか？
それとも、　AWS を使って仕事をするときの意味でしょうか？

The phrase 'When I work at AWS' gives me the nuance of having been employed and working at AWS. Is that correct because I don't know your previous jobs? Or does it mean when you work using AWS services?

Reply:
@takkanm すみません。 When I work with AWS　です。